applications/luci-app-natmap/htdocs/luci-static/resources/view/natmap.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+'require form';
+'require fs';
+'require rpc';
+'require view';
+'require tools.widgets as widgets';
+
+var callServiceList = rpc.declare({
+	object: 'service',
+	method: 'list',
+	params: ['name'],
+	expect: { '': {} }
+});
+
+function getInstances() {
+	return L.resolveDefault(callServiceList('natmap'), {}).then(function(res) {
+		try {
+			return res.natmap.instances || {};
+		} catch (e) {}
+		return {};
+	});
+}
+
+function getStatus() {
+	return getInstances().then(function(instances) {
+		var promises = [];
+		var status = {};
+		for (var key in instances) {
+			var i = instances[key];
+			if (i.running && i.pid) {
+				var f = '/var/run/natmap/' + i.pid + '.json';
+				(function(k) {
+					promises.push(fs.read(f).then(function(res) {
+						status[k] = JSON.parse(res);
+					}).catch(function(e){}));
+				})(key);
+			}
+		}
+		return Promise.all(promises).then(function() { return status; });
+	});
+}
+
+return view.extend({
+	load: function() {
+		return getStatus();
+	},
+	render: function(status) {
+		var m, s, o;
+
+		m = new form.Map('natmap', _('NATMap'));
+		s = m.section(form.GridSection, 'natmap');
+		s.addremove = true;
+		s.anonymous = true;
+
+		o = s.option(form.Flag, 'enable', _('Enable'));
+		o.editable = true;
+		o.modalonly = false;
+
+		o = s.option(form.ListValue, 'udp_mode', _('Protocol'));
+		o.default = '1';
+		o.value('0', 'TCP');
+		o.value('1', 'UDP');
+		o.textvalue = function(section_id) {
```

Comment:
I'm new to js, would you care to explain to me why this is needed?

Reply:
It will show `0` or `1` in grid without this.